### ISSUE TITLE
When (or not) to use `consider_prior` flag for TPE sampler

### ISSUE THREAD

**Issue by donny8** [optuna/optuna#3217]
I was following the [parzen_estimator](https://github.com/optuna/optuna/blob/1a4e46c48f3fbdcf6ce5cb23f12a6659dbcae92c/optuna/samplers/_tpe/parzen_estimator.py) for TPE sampler, and noticed `consider_prior` flag.

The flag seemed to consider prior (literally) so that the values originated for the high, low values are included in the [mus](https://github.com/optuna/optuna/blob/1a4e46c48f3fbdcf6ce5cb23f12a6659dbcae92c/optuna/samplers/_tpe/parzen_estimator.py#L450), [sigmas](https://github.com/optuna/optuna/blob/1a4e46c48f3fbdcf6ce5cb23f12a6659dbcae92c/optuna/samplers/_tpe/parzen_estimator.py#L219), [weights](https://github.com/optuna/optuna/blob/1a4e46c48f3fbdcf6ce5cb23f12a6659dbcae92c/optuna/samplers/_tpe/parzen_estimator.py#L219) (numerical params for example)

My question is, why the `consider_prior` flag is needed? When should I turn-on or turn-off the flag?

In other words, what was the motivation of this [update](https://github.com/optuna/optuna/pull/268)?
```
Currently, the sigma values of prior distributions are changed depending on the observations.
In this change, the sigma values are overwritten by a predetermined value, similarly to the TPE implementation of hyperopt.
```
Isn't it good to change the sigma values depending on the observations?

I think  overwriting by a predetermined value might rather disturb estimating the correct density especially when the number of observations is low.

**Comment by nzw0301** [comment]
@HideakiImamura could you take a look at this issue because you are very familiar with TPE sampler? 

Related to this issue, I suppose that it might be great to describe what happens with `consider_prior=False` in the docstring of `consider_prior` of [`TPESampler`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.TPESampler.html).

**Comment by HideakiImamura** [comment]
@nzw0301  Thanks for the notification.

In my understanding, the consider_prior flag is used to increase the exploration of TPE. Without the prior, TPE chooses the next parameter according to the density ratio. It is highly biased to the exploitation. To balance the exploration-exploitation trade-off, we (and original hyperopt's implementation) use the consider_prior flag.

In more detail, the "prior" to be considered is a probability distribution spread thinly over the entire search space. The each density is defined as a mixed Gaussian distribution. The prior is one of the mixed Gaussian. To increase the exploration, we only consider the information of the search space, not observed data.
